plenum/test/view_change/slow_nodes/test_view_change_gc_all_nodes_random_delay.py

```diff
@@ -0,0 +1,127 @@
+from itertools import combinations
+
+from stp_core.loop.eventually import eventually
+
+from plenum.test import waits
+from plenum.test.helper import sendRandomRequests, \
+        send_reqs_to_nodes_and_verify_all_replies, \
+        waitForSufficientRepliesForRequests, \
+        checkViewNoForNodes, \
+        check_last_ordered_3pc
+from plenum.test.delayers import delay_3pc_messages, \
+        reset_delays_and_process_delayeds
+from plenum.test.view_change.helper import ensure_view_change_complete
+from plenum.test.pool_transactions.conftest import clientAndWallet1, \
+        client1, wallet1, client1Connected, looper
+
+
+TestRunningTimeLimitSec = 300
+
+
+def ensure_nodes_last_ordered_3pc(nodes, last_ordered_3pc):
+    for n1, n2 in combinations(nodes, 2):
+        lst_3pc = check_last_ordered_3pc(n1, n2)
+
+    assert lst_3pc == last_ordered_3pc
+
+
+def ensure_nodes_requests_size(nodes, size):
+    for node in nodes:
+        assert len(node.requests) == size
+
+
+def test_view_change_gc_in_between_3pc_all_nodes_random_delays(
+        looper, txnPoolNodeSet, wallet1, client):
+    """
+    Test that garbage compares the whole 3PC key (viewNo, ppSeqNo) and
```

Comment:
@lovesh 
> View change causes GC so if you are in the same view you only need pp_seq_no.
And what if you are in the next view and you have not ordered but already created 3pc messages in queues (e.g. sentPrePrepares)?

Let's say we ordered 1 message in view 0: last_ordered = (0,1). View change happened.
In the next viewNo=1 we sent new message, 3pc batch (1,1) was formed but view change happened again before it was ordered. last_ordered is still (0,1), so GC will remove that batch because it's ppSeqNo not more than last_ordered.ppSeqNo.
For now GC ignores viewNo and relative to view_change, but last_ordered_3pc is not.